### PR TITLE
[Enhancement] Refactor functions in int_exp by templates

### PR DESCRIPTION
**Pull request by PragmaTwice**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

The current implementation of functions in `int_exp.h` used some hand-written, hard-coded arrays, which made the code lengthy. 
These functions can be rewritten using templates, which is more generalized and less error-prone.

Supervised by @jackwener. :kissing_heart:

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (?)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

### CODE REVIEW

**Review by BiteTheDDDDt — DISMISSED** [review]
LGTM

**Review by jackwener — DISMISSED** [review]
LGTM cc @yiguolei

**Comment by PragmaTwice** [comment]
Could you please review it again since I formatted the code? @BiteTheDDDDt

File: be/src/vec/common/int_exp.h [hunk 1]
```diff
@@ -22,6 +22,34 @@
 
 #include <cstdint>
 #include <limits>
+#include <utility>
+
+namespace exp_details {
+
+// compile-time exp(v, n) by linear recursion
+template <typename T, T v, std::size_t n>
+constexpr inline const T exp = v* exp<T, v, n - 1>;
```
Comment (BiteTheDDDDt) [inline be/src/vec/common/int_exp.h:31]: Seems here have a bug of clang-format, maybe we can use `T(v) * exp<T, v, n - 1>` to work around.

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

File: be/src/vec/common/int_exp.h (hunk 1 above)
Comment (PragmaTwice) [reply be/src/vec/common/int_exp.h]: done

**Pull request merged**